src/redux/reducer.js

```diff
@@ -904,8 +906,187 @@ export const reducer = createReducer(initialState, {
+                equipmentToDeleteId
+            );
+        }
+    },
+    [RESET_EQUIPMENTS]: (state) => {
+        state.spreadsheetNetwork = { ...initialSpreadsheetNetworkState };
+    },
+    [RESET_EQUIPMENTS_POST_LOADFLOW]: (state) => {
+        state.spreadsheetNetwork = {
+            ...initialSpreadsheetNetworkState,
+            substations: state.spreadsheetNetwork.substations,
+            voltageLevels: state.spreadsheetNetwork.voltageLevels,
+            hvdcLines: state.spreadsheetNetwork.hvdcLines,
+        };
+    },
 });
 
+function updateSubstationAfterVLDeletion(currentSubstations, VLToDeleteId) {
+    const substationToUpdateIndex = currentSubstations.findIndex((sub) =>
+        sub.voltageLevels.some((vl) => vl.id === VLToDeleteId)
+    );
+    if (substationToUpdateIndex >= 0) {
+        currentSubstations[substationToUpdateIndex].voltageLevels =
+            currentSubstations[substationToUpdateIndex].voltageLevels.filter(
+                (vl) => vl.id !== VLToDeleteId
+            );
+    }
+
+    return currentSubstations;
+}
+
+function fromEquipmentTypeToSpreadsheetNetworkType(equipmentType) {
+    switch (equipmentType) {
+        case EQUIPMENT_TYPES.LINE.type:
+            return 'lines';
+        case EQUIPMENT_TYPES.TWO_WINDINGS_TRANSFORMER.type:
+            return 'twoWindingsTransformers';
+        case EQUIPMENT_TYPES.THREE_WINDINGS_TRANSFORMER.type:
+            return 'threeWindingsTransformers';
+        case EQUIPMENT_TYPES.GENERATOR.type:
+            return 'generators';
+        case EQUIPMENT_TYPES.LOAD.type:
+            return 'loads';
+        case EQUIPMENT_TYPES.BATTERY.type:
+            return 'batteries';
+        case EQUIPMENT_TYPES.DANGLING_LINE.type:
+            return 'danglingLines';
+        case EQUIPMENT_TYPES.HVDC_LINE.type:
+            return 'hvdcLines';
+        case EQUIPMENT_TYPES.LCC_CONVERTER_STATION.type:
+            return 'lccConverterStations';
+        case EQUIPMENT_TYPES.VSC_CONVERTER_STATION.type:
+            return 'vscConverterStations';
+        case EQUIPMENT_TYPES.SHUNT_COMPENSATOR.type:
+            return 'shuntCompensators';
+        case EQUIPMENT_TYPES.STATIC_VAR_COMPENSATOR.type:
+            return 'staticVarCompensators';
+        case EQUIPMENT_TYPES.VOLTAGE_LEVEL.type:
+            return 'voltageLevels';
+        case EQUIPMENT_TYPES.SUBSTATION.type:
+            return 'substations';
+        default:
+            return;
+    }
+}
+
+function deleteEquipment(currentEquipments, equipmentToDeleteId) {
+    const equipmentToDeleteIndex = currentEquipments.findIndex(
+        (eq) => eq.id === equipmentToDeleteId
+    );
+    if (equipmentToDeleteIndex >= 0) {
+        currentEquipments.splice(equipmentToDeleteIndex, 1);
+    }
+    return currentEquipments;
+}
+
+function updateSubstationsAndVoltageLevels(
+    currentSubstations,
+    currentVoltageLevels,
+    newOrUpdatedSubstations
+) {
+    currentSubstations = updateEquipments(
```

Comment:
You shouldn't create a new const instead and rename it ? Same for VL ?

Reply:
fixed